From f864490943d596b1d1235bdd20ee5dc118b9958b Mon Sep 17 00:00:00 2001
From: "xiaoya.lin" <xiaoya.lin@amlogic.com>
Date: Wed, 19 Jul 2023 18:31:34 +0800
Subject: [PATCH 4/4] multienc: Support rgb888 [1/1]

PD#SWPL-132055

Problem:
Support rgb888

Solution:
Support rgb888

Verify:
Local

Signed-off-by: xiaoya.lin <xiaoya.lin@amlogic.com>
Change-Id: I47667f9e70eaf5ac6e3e80959c32c7df10bd2312
---
 multiEnc/amvenc_lib/AML_MultiEncoder.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/multiEnc/amvenc_lib/AML_MultiEncoder.c b/multiEnc/amvenc_lib/AML_MultiEncoder.c
index 8865dea..e6f21d5 100755
--- a/multiEnc/amvenc_lib/AML_MultiEncoder.c
+++ b/multiEnc/amvenc_lib/AML_MultiEncoder.c
@@ -69,11 +69,8 @@ static unsigned long long total_encode_frames;
 static struct timeval start_test;
 static struct timeval end_test;
 #endif
-#if defined(__ANDROID__)
+
 #define SUPPORT_SCALE 1
-#else
-#define SUPPORT_SCALE 0
-#endif
 
 #if SUPPORT_SCALE
 #include <ge2d_port.h>
@@ -1671,7 +1668,7 @@ AMVEnc_Status AML_MultiEncSetInput(amv_enc_handle_t ctx_handle,
         //memcpy((void *) ((char *)ctx->amlge2d.ge2dinfo.src_info[0].vaddr[0] + input->pitch * input->height), (void *)input->YCbCr[1], input->pitch * input->height / 4);
         //memcpy((void *) ((char *)ctx->amlge2d.ge2dinfo.src_info[0].vaddr[0] + (input->pitch * input->height * 5) /4), (void *)input->YCbCr[2], input->pitch * input->height / 4);
       } else if (ctx->fmt == AMVENC_RGB888) {
-        //memcpy((void *)ctx->amlge2d.ge2dinfo.src_info[0].vaddr[0], (void *)input->YCbCr[0], input->pitch * input->height * 3);
+        memcpy((void *)ctx->amlge2d.ge2dinfo.src_info[0].vaddr[0], (void *)input->YCbCr[0], input->pitch * input->height * 3);
       }
 
       do_strechblit(&ctx->amlge2d.ge2dinfo, input);
-- 
2.25.1

